bot/cogs/birthday/birthday_commands.py

```diff
@@ -0,0 +1,118 @@
+        self.bot = bot
+        self.post_birthdays.start()
+    
+    # Every day at 9AM Pacific Time
+    @tasks.loop(time=time(hour=17, minute=0, second=0, tzinfo=timezone.utc))
+    async def post_birthdays(self):
+        if not self.bot.is_ready():
+            return
+
+        birthdays = birthday_helper.get_todays_birthdays()
+        current_guild_id = None
+        birthday_channel_id = None
+        message = None
+        guild = None
+        for birthday in birthdays:
+            if birthday.guild_id != current_guild_id:
+                if message is not None and birthday_channel_id is not None:
+                    message = messages.get_special_birthday_fields(message)
+                    channel = self.bot.get_channel(birthday_channel_id)
+                    await channel.send(embed=message)
+                    message = None
+                guild = self.bot.get_guild(birthday.guild_id)
+                if guild is None:
+                    continue
+                current_guild_id = birthday.guild_id
+                birthday_channel_id = birthday_helper.get_birthday_channel_id(current_guild_id)
+                if birthday_channel_id is None:
+                    continue
+                message = messages.birthday_message()
+                member = guild.get_member(birthday.user_id)
+                message = messages.birthday_entry(message, birthday, member)
+            else:
+                member = guild.get_member(birthday.user_id)
+                message = messages.birthday_entry(message, birthday, member)
+        if message is not None and birthday_channel_id is not None: 
+            message = messages.get_special_birthday_fields(message)
+            channel = self.bot.get_channel(birthday_channel_id)
+            await channel.send(embed=message)
+
+
+    @post_birthdays.error
+    async def post_birthdays_error(self, e):
+        sentry_sdk.capture_exception(e)
+        await asyncio.sleep(60)
+        self.post_birthdays.restart()
+
+    @slash_command(name='birthday', guild_ids=[TESTING_GUILD_ID, BUMPERS_GUILD_ID],
+                   description='Birthday commands')
+    async def birthday(self, interaction: Interaction):
+        pass
+
+    @birthday.subcommand(name='add', description='Add a birthday')
+    async def birthday_add(self, interaction: Interaction,
+                            user: Member = SlashOption(name='user',
+                                                         description='User to add birthday for.'),
+                            name: str = SlashOption(name='name', description='Name of person celebrating a birthday'),
+                            date: str = SlashOption(name='date', description='Date of birthday (MM/DD/YYYY)')):
+        if interaction.guild is not None and interaction.guild_id not in [BUMPERS_GUILD_ID, TESTING_GUILD_ID]:
+            return await interaction.send(embed=messages.error('Birthday tracking is unsupported in this server.'))
```

Comment:
This actually isn't necessary in this command, since we're passing `guild_ids` to the `slash_command` decorator :) I had to do this in Secret Santa because those commands need to be usable in DMs with Angela, which means they have to be registered globally.

Since we're specifying the guild IDs in the definition of this slash command group, these commands won't be registered outside the pair of guilds we've passed to it.